File: leetcode/problems/14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#


# @lc code=start
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        prefix = ""

        strs.sort(key=lambda x: len(x))
        target = strs[0]

        for i in range(len(target)):
            char = target[i]
            for str in strs:
                if char != str[i]:
                    return prefix
            prefix += char

        return prefix


# @lc code=end

File: leetcode/problems/14.longest-common-prefix.java
/*
 * @lc app=leetcode id=14 lang=java
 *
 * [14] Longest Common Prefix
 */

// @lc code=start

import java.util.Arrays;

class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix = new String();
        Arrays.sort(strs);
        String target = strs[0];

        for (int i = 0; i < target.length(); i++) {
            for (String str : strs) {
                if (str.charAt(i) != target.charAt(i)) {
                    return prefix;
                }
            }

            prefix += target.charAt(i);
        }

        return prefix;
    }
}
// @lc code=end
